### ISSUE TITLE
coadd-redshifts script cframes glob getting temp files too

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1836]
The $CFRAMES glob in the coadd-redshifts*.slurm scripts is e.g.
```
CFRAMES=$(ls exposures/20201220/00069042/cframe-[brz]$SPECTRO-00069042.fits*)
```
That was intended to work with either gzipped or not input cframes, but it is also picking up temporary files leftover from crashes, e.g.
```
cframe-b1-00069042.fits_tmp42870.gz
```

Two items:
* Make a more specific glob
* Update the `desispec.io.util.get_tempfilename` to special-case the `.gz` extension so that this tempfile would have been `cframe-b1-00069042_tmp42870.fits.gz` instead which would not have passed the original glob.
  * context: `get_tempfilename` is creating a temporary filename in the same directory while preserving the extension so that code like `Table.write(filename)` can still auto-derive that this is a fits file.  In this case the extension is effectively `.fits.gz` not just `.gz`.

**Comment by akremin** [comment]
I am not a glob expert, but I am able to get `CFRAMES=$(ls exposures/20201220/00069042/cframe-[brz]$SPECTRO-00069042.fits{,.gz} 2>/dev/null)` to work reliably on the command line. I don't like that I need to redirect the error outputs, but that is required with this solution since one of the two curly bracket evaluations will always fail.

Let me know if that is acceptable or if you can come up with something better and I can implement this @sbailey .

**Comment by sbailey** [comment]
@akremin good solution, go for it, thanks

**Comment by sbailey** [comment]
implemented in PR #1853 .  Closing issue.
